build.gradle.kts

```diff
@@ -1,66 +1,81 @@
+@file:OptIn(ExperimentalKotlinGradlePluginApi::class)
+
 import io.github.petertrr.configurePublishing
+import io.github.petertrr.ext.booleanProperty
 import io.gitlab.arturbosch.detekt.Detekt
-import org.jetbrains.kotlin.gradle.targets.jvm.tasks.KotlinJvmTest
+import org.jetbrains.kotlin.gradle.ExperimentalKotlinGradlePluginApi
+import org.jetbrains.kotlin.gradle.dsl.JvmTarget
+import org.jetbrains.kotlin.gradle.dsl.KotlinVersion
 
 plugins {
     kotlin("multiplatform")
-    id("jacoco-convention")
     alias(libs.plugins.detekt)
+    id("jacoco-convention")
 }
 
 group = "io.github.petertrr"
 description = "A multiplatform Kotlin library for calculating text differences"
 
+dependencies {
+    detektPlugins(libs.detekt.formatting)
+}
+
 kotlin {
     explicitApi()
 
-    jvm()
-    js(IR) {
+    compilerOptions {
+        apiVersion = KotlinVersion.KOTLIN_1_9
+        languageVersion = KotlinVersion.KOTLIN_1_9
+    }
+
+    jvm {
+        compilations.configureEach {
+            compilerOptions.configure {
+                // Minimum bytecode level is 52
+                jvmTarget = JvmTarget.JVM_1_8
+
+                // Output interfaces with default methods
+                freeCompilerArgs.add("-Xjvm-default=all")
```

Comment:
This flag is necessary for Kotlin to output bytecode that uses JVM's interface default methods.  

For
```kotlin
interface Example {
  fun something(): String = 
    "example"
}
```

Kotlin by default will generate an additional class with an additional method.  
This means additional bytecode and call stack depth for no added value, since Java 8 supports default methods.

Unfortunately that flag isn't turned on by default yet, not sure why, probably for backward binary compatibility.
But we don't have a binary compatibility issue as this library isn't considered stable yet.